projects/grange-core/src/lib/authentication.service.ts

```diff
@@ -10,245 +10,274 @@ import { tap, catchError, map } from 'rxjs/operators';
-            .pipe(
-                catchError(this.error.bind(this))
-            );
-    }
+  requestPasswordReset(login: string): Observable<any> {
+    const headers = this.getHeaders();
+    const url =
+      this.config.get('BACKEND_URL') + `/@users/${login}/reset-password`;
+    return this.http
+      .post(url, {}, { headers: headers })
+      .pipe(
+        catchError(this.error.bind(this))
+      );
+  }
 
-    getHeaders(): HttpHeaders {
-        let headers = new HttpHeaders();
-        headers = headers.set('Accept', 'application/json');
-        headers = headers.set('Content-Type', 'application/json');
-        const auth = localStorage.getItem('auth');
-        if (auth) {
-            headers = headers.set('Authorization', 'Bearer ' + auth);
-        } else if (!!this.basicCredentials) {
-            headers = headers.set('Authorization', 'Basic ' + btoa(this.basicCredentials.join(':')));
-        }
-        return headers;
+  passwordReset(resetInfo: PasswordResetInfo): Observable<any> {
+    const headers = this.getHeaders();
+    const data: { [key: string]: string } = {
+      new_password: resetInfo.newPassword,
+    };
+    if (resetInfo.oldPassword) {
+      data['old_password'] = resetInfo.oldPassword;
     }
-
-    setAuthenticated(isAuthenticated: boolean) {
-        this.isAuthenticated.next({ state: isAuthenticated, pending: false, username: this.getUsername() });
+    if (resetInfo.token) {
+      data['reset_token'] = resetInfo.token;
     }
+    const url =
+      this.config.get('BACKEND_URL') +
+      `/@users/${resetInfo.login}/reset-password`;
+    return this.http
+      .post(url, data, { headers: headers })
+      .pipe(
+        catchError(this.error.bind(this))
+      );
+  }
 
-    protected error(errorResponse: HttpErrorResponse): Observable<Error> {
-        const error: Error = getError(errorResponse);
-        return Observable.throw(error);
+  getHeaders(): HttpHeaders {
+    let headers = new HttpHeaders();
+    headers = headers.set('Accept', 'application/json');
+    headers = headers.set('Content-Type', 'application/json');
+    const auth = localStorage.getItem('auth');
+    if (auth) {
+      headers = headers.set('Authorization', 'Bearer ' + auth);
+    } else if (!!this.basicCredentials) {
+      headers = headers.set('Authorization', 'Basic ' + btoa(this.basicCredentials.join(':')));
     }
+    return headers;
+  }
+
+  setAuthenticated(isAuthenticated: boolean) {
+    this.isAuthenticated.next({ state: isAuthenticated, pending: false, username: this.getUsername() });
+  }
+
+  protected error(errorResponse: HttpErrorResponse): Observable<Error> {
+    const error: Error = getError(errorResponse);
+    return Observable.throw(error);
+  }
+
+  goSocialLogin(provider, callback) {
+    const callUrl = `${this.config.get('BACKEND_URL')}/@authenticate/${provider}?callback=${location.origin}/${callback}${provider}`;
+    window.location.href = callUrl;
+  }
+
+  getValidationSchema(token) {
```

Comment:
it would be better to type the returned value:
```
getValidationSchema(token): Observable<boolean>
```
(assuming it is a boolean, and that's the whole point, if you type it, I know what Guillotina is supposed to return here)

Reply:
In this case its a dynamic json schema so would be any